src/zino/api/legacy.py

```diff
@@ -235,13 +235,15 @@ async def do_quit(self):
         self.transport.close()
 
     async def do_help(self):
-        responders = self._get_all_responders()
-        if not self.is_authenticated:
-            responders = {
-                name: func for name, func in responders.items() if not getattr(func, "requires_authentication", False)
-            }
-
-        commands = " ".join(sorted(responders))
+        """Lists all available top-level API commands"""
+        eligible = (
+            responder.name
+            for responder in self._responders.values()
+            if " " not in responder.name
+            and (self.is_authenticated or not getattr(responder.function, "requires_authentication", False))
+        )
```

Comment:
I would prefer this ultra long condition to be split into three parts with descriptive names, because it took me 5 minutes to wrap my head around this

Reply:
If you consider three expressions in boolean logic to be *ultra* long, you will have issues next time you look at legacy code :grin:

Reply:
Maybe I was exaggerating a little :grin:
But the nice thing about this re-implementation is that it doesn't have to be like legacy code :smile:

Reply:
Clarifications and test addition ready for review :)